### PR TITLE
Telemetry Packet Extraction

### PR DESCRIPTION
**Pull request by JackCotter**
Telemetry now extracts all info from packets including:
1. Callsign
2. Altitude
3. Latitude
4. Longitude

### CODE REVIEW

**Comment by JackCotter** [comment]
This code currently has a bug where the rust server will send 2 packets back to back very quickly. This is because the front end is making 2 connections. This should be fixed in the Rocket Monitoring Refactor PR (#76).

File: services/telemetry/src/telemetry.rs [hunk 1]
```diff
@@ -67,11 +86,30 @@ pub async fn start_telemetry(state_ref: Arc<Mutex<State>>) -> io::Result<()> {
...
-        let mut b = child_out.read(&mut readbuf);
-        extract_packets(Arc::clone(&state_ref), readbuf.clone()).await;
+        let mut bytes_read = child_out.read(&mut readbuf);
+        match bytes_read {
+            Ok(value) => {
+                let int_bytes_read: i32 = value as i32;
+                if (int_bytes_read > 0) {
+                    buffer.extend_from_slice(&readbuf[..value]);
+                    // Check for two successive newlines
+                    if let Some(pos) = buffer.windows(2).position(|window| window == b"\n\n") {
```
Comment (klemie) [inline services/telemetry/src/telemetry.rs:100]: when I run the code it couldn't find two consecutive new lines. We should test this with the brb on Thursday 

I did this before the if statement and it worked fine with the mock data
```rs
 extract_packets(Arc::clone(&state_ref), buffer.clone()).await;
 buffer.clear();
```

**Review by klemie — CHANGES_REQUESTED** [review]

**Comment by klemie** [comment]
@PokeyOne Can you test this in your environment. WSL couldn't pick up `\n` in the packets for some reason, wondering if mac can. Might need to find a better bay detect locked packets.

File: services/telemetry/src/telemetry.rs (hunk 1 above)
Comment (klemie) [reply services/telemetry/src/telemetry.rs:100]: @JackCotter Could we add a `Locked` attribute to the packet so we can more easily tell on the frontend if we've got a lock or not

**Pull request merged**